client/src/lib/Sources/SourceForm.svelte

```diff
@@ -27,12 +29,80 @@
     await loadCerts();
   };
 
+  export const fillAgeDataFromSource = (useSource: Source) => {
+    ageUnit = AgeUnit.hours;
+    let baseNumber: number | undefined = undefined;
+    let baseUnit: AgeUnit = ageUnit;
+    if (useSource.age && !["0s", "0h"].includes(useSource.age)) {
+      let [numStr, ...r]: string[] = (useSource.age ?? "").split("h");
+      let num: number = +numStr;
+      if (!(Number.isInteger(num) && !/[1-9]/.test(r.join(``)))) {
+        throw Error(
+          "Expected age to be given exclusively in hours, actual value was '" + useSource.age + "'."
+        );
+      }
+      if (num && num % (24 * 7) === 0) {
+        baseNumber = num / (24 * 7);
+        baseUnit = AgeUnit.weeks;
+      } else if (num && num % 24 === 0) {
+        baseNumber = num / 24;
+        baseUnit = AgeUnit.days;
+      } else {
+        baseNumber = num;
+        baseUnit = AgeUnit.hours;
+      }
+    } else if (useSource.age && ["0s", "0h"].includes(useSource.age)) {
+      baseNumber = 0;
+    }
+    ageNumber = baseNumber;
+    ageUnit = baseUnit;
+  };
+
   export let inputChange = () => {};
 
+  enum AgeUnit {
+    hours = "h",
+    days = "d",
+    weeks = "w"
+  }
+
+  const ageUnits = [
+    { value: AgeUnit.hours, name: "hours" },
+    { value: AgeUnit.days, name: "days" },
+    { value: AgeUnit.weeks, name: "weeks" }
+  ];
+
   let headers: [string, string][] = [["", ""]];
   let privateCert: FileList | undefined;
   let publicCert: FileList | undefined;
 
+  let ageUnit: AgeUnit;
+  let ageNumber: number | undefined;
+  let previousAgeNumber: number | undefined;
+
+  onMount(() => {
+    fillAgeDataFromSource(source);
+  });
```

Comment:
```Typescript
$: if (source) {
   // Do things that parse input from `source`
}
```
`onMount` will only be called only once. The source editor updates `source` after saving the source; refilling the age data may be required.

Reply:
I haven't done this here because source.age would get overwritten every time the value in the inputs is changed.
Instead I made this dependant on the calling Component, from which `fillAgeDatafromSource` can be called, every time source was actually changed (see the changes in `SourceEditor.svelte`).